### PR TITLE
Add GitHub Action for building and pushing UI and BFF images

### PR DESCRIPTION
**Pull request by Griffin-Sullivan**
## Description
Added a GH Action for building and pushing the UI and BFF images. Reuses the build script for Model Registry server, and edits the Makefile a little bit to get everything working. I also turned off image builds for MR server when a PR is being made only for UI / BFF changes. I can remove that if we want to keep building them.

## How Has This Been Tested?
Going to test the PR as `on: pull_request` then if it passes will update the PR to run the action `on: push`.

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [x] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

If you have UI changes

- [ ] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

**Comment by Griffin-Sullivan** [comment]
My user must not have access to the creds for the kubeflow docker so I'm assuming there's no way to test this without having the bot do it on push to main? @tarilabs any ideas how I can test?

File: .github/workflows/build-and-push-ui-images.yml
Comment (tarilabs) [inline .github/workflows/build-and-push-ui-images.yml:3]: I appreciate in the PR description is mentioned this `on: pull_request` is only for testing, but I'd rather make a comment so to make sure we don't forget before merging 😅

**Review by tarilabs — CHANGES_REQUESTED** [review]
> My user must not have access to the creds for the kubeflow docker so I'm assuming there's no way to test this without having the bot do it on push to main? @tarilabs any ideas how I can test?

When I face similar situation @Griffin-Sullivan , I did setup `DOCKERHUB_USERNAME` etc in my GitHub repo (my fork) and updated `IMG_ORG` to point at my username in dockerhub.io

It also gives you an idea/survey of the changes we will need to apply midstream (kind request: please remind me/us when this PR will be synchronized midstream!)

**Comment by Griffin-Sullivan** [comment]
You can see on https://github.com/Griffin-Sullivan/model-registry/actions/runs/11147635638/job/30982430523?pr=1 that the Action is passing when I push to my own quay repo. You can find them here: https://quay.io/user/gsulliva/?tab=repos

The reason the build-and-test-image action is failing is because I'm forcing the build script to run with `quay.io` so once that is removed when we are ready to merge it should pass fine.

File: .github/workflows/build-and-push-ui-images.yml
Comment (alexcreasy) [inline .github/workflows/build-and-push-ui-images.yml:21]: Just an extra reminder this line is here :D

File: scripts/build_deploy.sh
Comment (alexcreasy) [inline scripts/build_deploy.sh:6]: Another reminder :D

**Comment by alexcreasy** [comment]
Looks good, I haven't completely groked the logic but the general tasks all look good!

**Comment by Griffin-Sullivan** [comment]
Latest run is still passing on https://github.com/Griffin-Sullivan/model-registry/actions/runs/11162537114/job/31027401589?pr=1. If someone will review this I can remove all the TODOs :smile:

File: .github/workflows/build-and-push-ui-images.yml
Comment (tarilabs) [inline .github/workflows/build-and-push-ui-images.yml:51]: this `IMG` will need to be prefixed with `quay.io` for midstream, correct?

I'm thinking if we should just be explicit and introduce `docker.io/` prefix in all GHAs 🤔 
what is your team thought here?:

**Review by tarilabs — COMMENTED** [review]
I think it's worthy to remove the TODOs

/lgtm

**Review by ederign — APPROVED** [review]

**Comment by ederign** [comment]
@tarilabs we need you to merge this one!

**Review by alexcreasy — APPROVED** [review]
All looks good to me.

**Comment by alexcreasy** [comment]
@tarilabs would you mind approving this, if you're happy with the changes? It's outside my or Eder's jurisdiction :)

**Review by tarilabs — APPROVED** [review]
- https://github.com/kubeflow/model-registry/pull/446#pullrequestreview-2349523222
- https://github.com/kubeflow/model-registry/pull/446#pullrequestreview-2354408068
- https://github.com/kubeflow/model-registry/pull/446#pullrequestreview-2354417148

/approve

**Comment by alexcreasy** [comment]
thanks @tarilabs

**Pull request merged**
